File: AD-HOC/2457/main.dart
import 'dart:io';

double percentage(String char, List<String> text) {
  int count = 0;
  for (String word in text) {
    if (word.contains(char)) count++;
  }
  return 100.0 * count / text.length;
}

void main() {
  String char = stdin.readLineSync()!.trim();
  List<String> text = stdin.readLineSync()!.trim().split(' ');
  print(percentage(char, text).toStringAsFixed(1));
}

File: AD-HOC/2457/Main.java
import java.util.Scanner;

public class Main {

    static double percentage(String character, String[] text) {
        int count = 0;
        for (String word : text)
            if (word.contains(character))
                count++;
        return 100.0 * count / text.length;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String character = scanner.nextLine().trim();
        String[] text = scanner.nextLine().trim().split(" ");
        System.out.printf("%.1f\n", percentage(character, text));
        scanner.close();
    }
}
